improver_tests/calibration/test_init.py

```diff
@@ -129,32 +144,107 @@ def test_probability_data_with_land_sea_mask(self):
         a single land-sea mask are provided, the groups are created as
         expected."""
 
-        forecast, truth, land_sea_mask = split_forecasts_and_truth(
+        (
+            forecast,
+            truth,
+            land_sea_mask,
+            additional_predictors,
+        ) = split_forecasts_and_truth(
             self.probability_forecasts + self.probability_truths + [self.landsea_mask],
             self.truth_attribute,
+            self.landsea_mask.name(),
+        )
+
+        self.assertIsInstance(forecast, iris.cube.Cube)
```

Comment:
I've converted the 8 tests which cover the intended functionality of `split_forecasts_and_truth()` into a new pytest test, with forecast type (either 'probability' or 'realization'), and whether or not to include a land-sea mask and whether or not to include an additional predictor each being parameterised. Some exception tests for this plugin are still implemented in the unittest framework as I didn't think that converting these would substantially reduce the amount of code required.

I also took a lot of the code in the test `setup()` function into a new function outside of the unittest class so that the functionality could be used within both the unitttest and pytest tests.